src/appleseed/renderer/modeling/aov/aov.h

```diff
@@ -80,6 +84,29 @@ class APPLESEED_DLLSYMBOL AOV
         const size_t index) const = 0;
 };
 
+
+//
+// ColorAOV base class.
+//
+
+class ColorAOV
+  : public AOV
+{
+  public:
+
+    // Constructor.
+    ColorAOV(const char* name, const ParamArray& params);
+
+    // Return the number of channels of this AOV.
+    size_t get_channel_count() const override;
```

Comment:
I wonder if this method (and possibly others) couldn't be made non-virtual by implementing it in the base, and letting derived classes pass these information to the base class during construction.

Reply:
The interface of AOVs is not finished yet. Making the methods non-virtual is something we could 
change a bit later when things are more clear.

For now, these methods are only used during AOV writting, where performance is not that important.